background.ts

```diff
@@ -519,37 +518,19 @@ Electron.ipcMain.handle('service-forward', async(event, service, state) => {
 });
 
 Electron.ipcMain.on('k8s-integrations', async(event) => {
-  if (k8smanager instanceof WSLBackend) {
-    event.reply('k8s-integrations', await k8smanager?.listIntegrations());
-  }
+  // listIntegrations triggers integration-update, no need to deal with results.
+  await integrationManager.listIntegrations();
 });
 
 Electron.ipcMain.on('k8s-integration-set', async(event, name, newState) => {
-  if (k8smanager instanceof WSLBackend) {
-    console.log(`Setting k8s integration for ${ name } to ${ newState }`);
-    writeSettings({ kubernetes: { WSLIntegrations: { [name]: newState } } });
-    const currentState = await k8smanager.listIntegrations();
-
-    if (!(name in currentState) || currentState[name] === newState) {
-      event.reply('k8s-integrations', currentState);
-
-      return;
-    }
-    if (typeof currentState[name] === 'string') {
-      // There is an error, and we cannot set the integration
-      event.reply('k8s-integrations', currentState);
-
-      return;
-    }
-    const error = await k8smanager.setIntegration(name, newState);
+  console.log(`Setting k8s integration for ${ name } to ${ newState }`);
+  writeSettings({ kubernetes: { WSLIntegrations: { [name]: newState } } });
+  // listIntegrations triggers integration-update, no need to deal with results.
+  await integrationManager.listIntegrations();
```

Comment:
This will end up emitting `integration-update` on `mainEvents` twice if I'm not mistaken. Once in `listIntegrations`, and once in the `listIntegrations` call at the end of `sync`.